### PR TITLE
Stage 4 PR for proposal-intl-extend-timezonename

### PR DESCRIPTION
**Pull request by FrankYFTang**
Based on https://tc39.es/proposal-intl-extend-timezonename/

The &lt;del&gt; and some &lt;ins&gt; in the spec text is already in latest ECMA402

pending on Stage 4 Advancement in 2021-Dec TC39

### CODE REVIEW

File: spec/datetimeformat.html
Comment (ryzokuken) [inline spec/datetimeformat.html:289]: nit: are the brackets necessary here?

File: spec/datetimeformat.html
Comment (ryzokuken) [inline spec/datetimeformat.html:292]: @FrankYFTang have you seen any other instances of this symbol being used? What about just writing down "is not equal to"?

**Review by ryzokuken — APPROVED** [review]
LGTM, a few nits inline.

BTW, the diff at https://tc39.es/proposal-intl-extend-timezonename/ is outdated around FormatDateTimePattern pt 15.e.

File: spec/datetimeformat.html
Comment (FrankYFTang) [reply spec/datetimeformat.html:292]: ~/tc39/ecma402$ egrep -n "1..*≠" spec/*.html
spec/datetimeformat.html:285:            1. Else if _optionsProp_ ≠ _formatProp_, then
spec/numberformat.html:695:        1. If _f_ ≠ 0, then
spec/pluralrules.html:51:         1. If _f_ ≠ 0, then

**Comment by FrankYFTang** [comment]
> LGTM, a few nits inline.
> 
> BTW, the diff at https://tc39.es/proposal-intl-extend-timezonename/ is outdated around FormatDateTimePattern pt 15.e.

I know, I landed https://github.com/tc39/ecma402/commit/dbe5d7253a3b9bb8c14128f91a6e3546119b0543#diff-aae61752e383a82db0847140fc85c3f1718f6348407153200b6db8462f10a21a myself. So the diff in this PR is only the diff from the spec text.

File: spec/datetimeformat.html
Comment (FrankYFTang) [reply spec/datetimeformat.html:289]: I think that make it clearer

**Comment by ryzokuken** [comment]
@FrankYFTang great, thanks. Consider my questions resolved.

@gibson042 @leobalter do you folks want to take a quick look?

Good luck for Stage 4!

File: spec/datetimeformat.html
Comment (gibson042) [inline spec/datetimeformat.html:301]: @FrankYFTang this works, but I would love for the adjustments to be expressed more concisely. What would you think of this?
```suggestion
            1. Else if _property_ is *"timeZoneName"*, then
              1. Let _penalty_ be the value of the column with name matching _formatProp_ in the row of <emu-xref href="#table-timezonename-component-non-default-penalties"></emu-xref> for which the value of the &ldquo;Requested Style&rdquo; column is _optionsProp_. If that cell is empty, let _penalty_ be _removalPenalty_.
              1. If _optionsProp_ = _formatProp_, then
                1. Assert: _penalty_ is 0.
              1. Else,
                1. Decrease _score_ by _penalty_.
```
```html
<emu-table id="table-timezonename-component-non-default-penalties">
  <emu-caption>*"timeZoneName"* component non-default penalties</emu-caption>
```

  <table class="real-table">
    <thead>
      <tr>
        <th>Requested Style</th>
        <th>*"short"*</th>
        <th>*"long"*</th>
        <th>*"shortOffset"*</th>
        <th>*"longOffset"*</th>
        <th>*"shortGeneric"*</th>
        <th>*"longGeneric"*</th>
      </tr>
    </thead>
    <tr>
      <td>*"short"*</td>
      <td>0</td>
      <td>_shortMorePenalty_</td>
      <td>_offsetPenalty_</td>
      <td>_offsetPenalty_ + _shortMorePenalty_</td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td>*"shortGeneric"*</td>
      <td></td>
      <td></td>
      <td>_offsetPenalty_</td>
      <td>_offsetPenalty_ + _shortMorePenalty_</td>
      <td>0</td>
      <td>_shortMorePenalty_</td>
    </tr>
    <tr>
      <td>*"shortOffset"*</td>
      <td></td>
      <td></td>
      <td>0</td>
      <td>_shortMorePenalty_</td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td>*"long"*</td>
      <td>_longLessPenalty_</td>
      <td>0</td>
      <td>_offsetPenalty_ + _longLessPenalty_</td>
      <td>_offsetPenalty_</td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td>*"longGeneric"*</td>
      <td></td>
      <td></td>
      <td>_offsetPenalty_ + _longLessPenalty_</td>
      <td>_offsetPenalty_</td>
      <td>_longLessPenalty_</td>
      <td>0</td>
    </tr>
    <tr>
      <td>*"longOffset"*</td>
      <td></td>
      <td></td>
      <td>_longLessPenalty_</td>
      <td>0</td>
      <td></td>
      <td></td>
    </tr>
  </table>
</emu-table>

**Review by gibson042 — APPROVED** [review]

File: spec/datetimeformat.html
Comment (FrankYFTang) [reply spec/datetimeformat.html]: I am nervous about this suggestion. This is a much bigger (pure editorial I assume) changes. How about we merge it w/o such into ECMA402 and then later put that into a different Editorial PR?

Comment (gibson042) [reply spec/datetimeformat.html]: I don't object either way. But for clarity: would you support such a change, or do you prefer the algorithm steps as they are?

Comment (FrankYFTang) [reply spec/datetimeformat.html]: Let me put down this way- I am supporting change it to a table form as an editorial change but I think we need more spend more time and review from more people (in particular @anba) for that PR to make sure we didn't change unintentionally. So I prefer not mix that into this Stage 4 PR.

**Review by ryzokuken — APPROVED** [review]
LGTM. I like the change proposed by @gibson042 but we can do it as an editorial ECMA-402 PR after merging this, as @FrankYFTang proposed.

**Review by gibson042 — APPROVED** [review]

**Comment by ryzokuken** [comment]
Since we've hit stage 4 and all editorial issues have been addressed, can we merge this?

**Comment by FrankYFTang** [comment]
> Since we've hit stage 4 and all editorial issues have been addressed, can we merge this?

@ryzokuken  Who are you asking this question to? Are you asking @gibson042 ? I do not have merge right since I am not editor so we need someone have merge right to click the button to merge.

**Comment by ryzokuken** [comment]
@FrankYFTang I have merge access, I just wanted to confirm that this PR is good to go before doing that.

**Comment by FrankYFTang** [comment]
> I just wanted to confirm that this PR is good to go before doing that.

Are you asking @gibson042 ? or are you asking me? or someone else?

**Comment by ryzokuken** [comment]
@FrankYFTang I was asking both of you what you think. But I believe we're good to go, so since there aren't any objections, I'll merge this.

**Comment by FrankYFTang** [comment]
thanks

**Pull request merged**
